src/panels/config/automation/condition/types/ha-automation-condition-time.ts

```diff
@@ -76,7 +76,7 @@ export class HaTimeCondition extends LitElement implements ConditionElement {
           >
             <ha-switch
               .day=${day.name}
-              .checked=${this._isChecked(weekday, day.name)}
+              .checked=${weekday ? weekday.includes(day.name) : false}
```

Comment:
```suggestion
              .checked=${weekday?.includes(day.name)}
```

Actually maybe this works even better

Reply:
done